src/transformers/trainer_tf.py

```diff
@@ -14,6 +14,23 @@
 from .training_args_tf import TFTrainingArguments
 
 
+try:
+    import wandb
+
+    wandb.ensure_configured()
+    if wandb.api.api_key is None:
+        _has_wandb = False
+        wandb.termwarn("W&B installed but not logged in.  Run `wandb login` or set the WANDB_API_KEY env variable.")
+    else:
+        _has_wandb = False if os.getenv("WANDB_DISABLED") else True
+except (ImportError, AttributeError):
+    _has_wandb = False
+
+
+def is_wandb_available():
+    return _has_wandb
```

Comment:
I'm ok with moving this block to trainer_utils (it can be in a following PR)

Reply:
Sounds good, I'll propose it in a follow-up PR because we still need to import wandb in the other modules so I need to think of a clean approach.